v2/datastream-to-spanner/terraform/samples/mysql-end-to-end/main.tf

```diff
@@ -6,6 +6,23 @@ locals {
   migration_id = var.common_params.migration_id != null ? var.common_params.migration_id : random_pet.migration_id.id
 }
 
+# Setup network firewalls for datastream if creating a private connection.
+resource "google_compute_firewall" "allow-datastream" {
+  depends_on  = [google_project_service.enabled_apis]
+  count       = var.datastream_params.private_connectivity != null ? 1 : 0
+  project     = var.common_params.host_project != null ? var.common_params.host_project : var.common_params.project
+  name        = "allow-datastream"
+  network     = var.common_params.host_project != null ? "projects/${var.common_params.host_project}/global/networks/${var.datastream_params.private_connectivity.vpc_name}" : "projects/${var.common_params.project}/global/networks/${var.datastream_params.private_connectivity.vpc_name}"
+  description = "Allow traffic from private connectivity endpoint of Datastream"
+
+  allow {
+    protocol = "tcp"
+    ports    = ["3306"]
+  }
+  source_ranges = [var.datastream_params.private_connectivity.range]
+  target_tags   = ["databases"]
+}
+
```

Comment:
1. For the firewall rule, how does the user configure what targets the rule should apply to? 
Depending on the existing user infra setup - this can mean different things - some may have an IP range to configure, others may have a tag etc. You should provide the user the ability to make this configuration.

2. Adding a firewall rule is a very sensitive operation from a security POV. Customers may not want to have this automated. Just like other existing optional options in the template around security sensitive things (refer to README e.g *adding permissions to service account*), you should also make this optional, so the user can safely skip it.